src/StackExchange.Metrics/Handlers/StatsdMetricHandler.cs

```diff
@@ -13,10 +13,104 @@
 
 namespace StackExchange.Metrics.Handlers
 {
+    /// <summary>
+    /// Implements <see cref="IMetricHandler" /> by sending data to a statsd endpoint.
+    /// </summary>
+    public class StatsdMetricHandler : IMetricHandler
+    {
+        string _host;
+        ushort _port;
+        IMetricHandler _activeHandler;
+
+        /// <summary>
+        /// Constructs a new <see cref="StatsdMetricHandler" /> pointing at the specified host and port.
+        /// </summary>
+        /// <param name="host">
+        /// Host of a statsd endpoint.
+        /// </param>
+        /// <param name="port">
+        /// Port of a stats endpoint.
+        /// </param>
+        /// <remarks>
+        /// If the host or port are set to <c>null</c>/<c>0</c> this handler will be deactivated (every operation is a no-op).
+        /// Otherwise it sends data to a StatsD endpoint via UDP.
+        /// </remarks>
+        public StatsdMetricHandler(string host, ushort port)
+        {
+            Host = host;
+            Port = port;
+        }
+
+        /// <summary>
+        /// Host to which we should send metrics to.
+        /// </summary>
+        public string Host
+        {
+            get => _host;
+            set
+            {
+                _host = value;
+                var oldHandler = _activeHandler;
+                if (oldHandler != null)
+                {
+                    oldHandler.Dispose();
+                }
+
+                _activeHandler = GetHandler();
+            }
+        }
+
+        /// <summary>
+        /// Port to which we should send metrics to.
+        /// </summary>
+        public ushort Port
+        {
+            get => _port;
+            set
+            {
+                _port = value;
+                var oldHandler = _activeHandler;
+                if (oldHandler != null)
+                {
+                    oldHandler.Dispose();
+                }
+
+                _activeHandler = GetHandler();
+            }
+        }
+
+        /// <inheritdoc />
+        public IMetricReadingBatch BeginBatch() => _activeHandler.BeginBatch();
+
+        /// <inheritdoc />
+        public void SerializeMetadata(IEnumerable<Metadata> metadata) => _activeHandler.SerializeMetadata(metadata);
+
+        /// <inheritdoc />
+        public void SerializeMetric(in MetricReading reading) => _activeHandler.SerializeMetric(reading);
+
+        /// <inheritdoc />
+        public ValueTask FlushAsync(TimeSpan delayBetweenRetries, int maxRetries, Action<AfterSendInfo> afterSend,
```

Comment:
For some reason, in `SignalFxMetricHandler` I added an elvis operator when accessing `_activeHandler` in `FlushAsync`; I suspect it might not be needed, but worth validating in tests I think

Reply:
Adding it for good measure. I don't see any path that would leave us with a `null` `_activeHandler` - nor how I'd trigger that in a test - but I see now harm returning gracefully from `FlushAsync` instead of throwing an NRE if that should ever be the case.